addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,230 @@ odoo.define('website_event.editor', function (require) {
+            useCurrent: false,
+            viewDate: moment(new Date()).hours(minDate.hours()).minutes(minDate.minutes()).seconds(minDate.seconds()).milliseconds(minDate.milliseconds()),
+            calendarWeeks: false,
+            icons: {
+                time: 'fa fa-clock-o',
+                date: 'fa fa-calendar',
+                next: 'fa fa-chevron-right',
+                previous: 'fa fa-chevron-left',
+                up: 'fa fa-chevron-up',
+                down: 'fa fa-chevron-down',
+            },
+            locale : moment.locale(),
+            allowInputToggle: true,
+        });
+        $dateGroup.on('error.datetimepicker', function (err) {
+            if (err.date) {
+                if (err.date < minDate) {
+                    Dialog.alert(this, _t('The date you selected is lower than the minimum date: ') + minDate.format(datetimepickerFormat));
+                }
+
+                if (err.date > maxDate) {
+                    Dialog.alert(this, _t('The date you selected is greater than the maximum date: ') + maxDate.format(datetimepickerFormat));
+                }
+            }
+            return false;
+        });
+    },
+    /**
+     * @private
+     */
+    _prepareFormValues: function (){
+        return {
+            name: this.$('input[name=name]').val().trim(),
+            location: this.$('select[name=event_location]').val(),
+            addressId: this.$('input[name=address_id]').val(),
+            eventStart: this.$('input[name=event_start]').val(),
+            eventEnd: this.$('input[name=event_end]').val(),
+        };
+    },
+    /**
+     * @private
+     * @param {*} values
+     * @param {*} errors
+     */
+    _validateDates: function (values, errors) {
+        if (!values.eventStart) {
+            errors.push('event_start');
+        } else if (!values.eventEnd) {
+            errors.push('invalid_end');
+        } else {
+            var momentStartDate = moment(values.eventStart);
+            var momentEndDate = moment(values.eventEnd);
+            if (!momentEndDate.isAfter(momentStartDate)) {
+                errors.push('invalid_end');
+            }
+        }
+    },
+    /**
+     * @private
+     * @param {*} values
+     */
+    _validateForm: function (values){
+        var errors = [];
+        if (!values.name){
+            errors.push('name');
+        }
+        if (values.location === 'on_site' && !values.addressId) {
+            errors.push('address');
+        }
+        this._validateDates(values, errors);
+        if (errors) {
+            this._display_errors(errors);
+        }
+        return errors.length === 0;
+    },
+    /**
+     * @private
+     */
+    _submitForm: function () {
+        var self = this;
+        var values = this._prepareFormValues();
+        if (!this._validateForm(values)){
+            return;
+        }
+        return self._rpc({
+        route: '/event/add_event',
+        params: {
```

Comment:
Couldn't prepare form values return a dict that we directly send to add_event ? That way we don't have to repeat keys in params.